src/common/qat.c

```diff
@@ -348,37 +393,54 @@ qat_dc_compress_async(
 			numBuffers,
 			&bufferMetaSize);
 
-	/* Allocate source buffer */
-	if (status == CPA_STATUS_SUCCESS)
-		status = mem_alloc_contig(
-				(void *)&pBufferMetaSrc, bufferMetaSize, 1);
+	if (!SVM_enabled) {
+		/** Allocate source buffer */
```

Comment:
Add a comment here - If SVM (Shared Virtual Memory) is not enabled, we need to allocate physical contiguous memory for input and output data and copy the data from/to the virtual memory.

Reply:
done